EzMap.Api/Controllers/PoiCollectionController.cs

```diff
@@ -46,37 +82,97 @@ public async Task<IActionResult> Search([FromServices] IUnitOfWork uow,
+                return Ok("No changes were made to the POI collection.");
+            }
 
-        if (dbPoiCol is not null) uow.PoiCollectionRepository.UpdatePoiCollectionAsync(dbPoiCol, dto);
+            var poiColUpdateIndex = new PoiCollectionUpdateIndexingModel(dto.Id, dto.Name, dto.Description);
+            var esResult = await elasticSearchService.AddOrUpdate(poiColUpdateIndex);
+            if (!esResult)
+            {
+                await transaction.RollbackAsync();
+                return StatusCode(StatusCodes.Status500InternalServerError,
+                    "Failed to index the POI collection. The operation has been rolled back.");
+            }
 
-        return await uow.SaveAsync() > 0
-            ? Ok("Your poi collection is updated successfully")
-            : new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            await transaction.CommitAsync();
+            return Ok("Your poi collection is updated successfully");
+        }
+        catch
+        {
+            await transaction.RollbackAsync();
+            throw;
+        }
     }
 
     [Authorize]
     [HttpDelete("{id:guid}")]
-    public async Task<IActionResult> Delete(Guid id, [FromServices] IUnitOfWork uow)
+    public async Task<IActionResult> Delete(Guid id, [FromServices] IUnitOfWork uow,
+        [FromServices] IElasticSearchService elasticSearchService)
     {
-        if (string.IsNullOrEmpty(id.ToString()))
+        await using var transaction = await uow.BeginTransactionAsync();
+
+        try
         {
-            return BadRequest("Please provide a valid id!");
-        }
+            if (string.IsNullOrEmpty(id.ToString()))
+            {
+                return BadRequest("Please provide a valid id!");
+            }
 
-        await uow.PoiCollectionRepository.DeletePoiCollectionAsync(id);
+            await uow.PoiCollectionRepository.DeletePoiCollectionAsync(id);
+            var dbResult = await uow.SaveAsync();
+            if (dbResult <= 0)
+            {
+                await transaction.RollbackAsync();
+                return NotFound("Poi collection not found.");
+            }
 
-        return await uow.SaveAsync() > 0
-            ? Ok("Your poi collection is deleted successfully!")
-            : new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            var esResult = await elasticSearchService.Remove(id.ToString());
+            if (!esResult)
+            {
+                await transaction.RollbackAsync();
+                return StatusCode(StatusCodes.Status500InternalServerError, new
+                {
+                    Message =
+                        "Failed to remove POI collection from ElasticSearch. The operation has been rolled back.",
+                });
+            }
+
+            await transaction.CommitAsync();
+            return Ok(new
+            {
+                Message = "Your poi collection is deleted successfully!",
```

Comment:
Same for inconsistency, this returns { Message }. Other places return plain string or other structure